cellar_rebalancer/src/collector/poller.rs

```diff
@@ -71,11 +88,61 @@ impl<T: 'static + Middleware> Poller<T> {
             },
             contract_state: UniswapV3CellarState::new(cellar.cellar_address, client),
             pool,
+            cosmos_key: rebalancer_config.load_deep_space_key(name.clone()),
         };
 
         Ok(poller)
     }
 
+    fn to_allocation(&self) -> proto::Allocation {
+        let tick_range: Vec<proto::TickRange> = self
+            .time_range
+            .tick_weights
+            .iter()
+            .map(|tick_weight| proto::TickRange {
+                upper: tick_weight.upper_bound as i32,
+                lower: tick_weight.lower_bound as i32,
+                weight: tick_weight.weight as u32,
+            })
+            .collect();
+
+        proto::Allocation {
+            vote: Some(proto::RebalanceVote {
+                cellar: Some(proto::Cellar {
+                    id: self.time_range.pair_id.to_string(),
+                    tick_ranges: self
+                        .time_range
+                        .tick_weights
+                        .iter()
+                        .map(|tick_weight| proto::TickRange {
+                            upper: tick_weight.upper_bound as i32,
+                            lower: tick_weight.lower_bound as i32,
+                            weight: tick_weight.weight as u32,
+                        })
+                        .collect(),
+                }),
+                current_price: self.contract_state.gas_price.unwrap().as_u64(),
+            }),
+            salt: "".to_string(), //TODO: Add salt,
```

Comment:
I think this is the last important thing we should be sure to update before merging. Not sure there's anything in the spec as to how large this should be but generating a random 8 character string using the OS randomness should be sufficient I'd imagine.

Reply:
Probably something along the lines of

```
use rand::{distributions::Alphanumeric, rngs::OsRng};

...

salt: OsRng
    .sample_iter(&Alphanumeric)
    .take(8)
    .map(char::from)
    .collect();
```